rcl_action/test/rcl_action/test_action_client.cpp

```diff
@@ -349,3 +351,50 @@ TEST_F(TestActionClientFixture, test_action_client_get_options) {
   options = rcl_action_client_get_options(&this->action_client);
   ASSERT_NE(options, nullptr) << rcl_get_error_string().str;
 }
+
+TEST_F(TestActionClientBaseFixture, test_action_client_init_fini_maybe_fail)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    int64_t count = rcutils_fault_injection_get_count();
+    rcl_node_t node = rcl_get_zero_initialized_node();
+    rcl_node_options_t node_options = rcl_node_get_default_options();
+    std::string node_name = std::string("test_action_client_node_") + std::to_string(count);
+    rcl_ret_t ret = rcl_node_init(&node, node_name.c_str(), "", &this->context, &node_options);
+    if (RCL_RET_OK != ret) {
+      continue;
+    }
+    const rosidl_action_type_support_t * action_typesupport = ROSIDL_GET_ACTION_TYPE_SUPPORT(
+      test_msgs, Fibonacci);
+    std::string action_name = std::string("test_action_client_name_") + std::to_string(count);
+    rcl_action_client_t action_client = rcl_action_get_zero_initialized_client();
+    rcl_action_client_options_t action_client_options = rcl_action_client_get_default_options();
+
+    ret = rcl_action_client_init(
+      &action_client,
+      &node,
+      action_typesupport,
+      action_name.c_str(),
+      &action_client_options);
+
+    if (RCL_RET_OK == ret) {
+      ret = rcl_action_client_fini(&action_client, &node);
```

Comment:
Should it be an extra `if` checking if fini completed properly?

Reply:
Seems reasonable. But with `rcl_action_client_fini` you can only reset the error message.